kibana/connecting_client.go

```diff
@@ -84,19 +85,24 @@ func NewConnectingClient(cfg *kibana.ClientConfig) Client {
 
 // Send tries to send a request to Kibana via established connection and returns unparsed response
 // If no connection is established an error is returned
-func (c *ConnectingClient) Send(method, extraPath string, params url.Values,
+func (c *ConnectingClient) Send(ctx context.Context, method, extraPath string, params url.Values,
 	headers http.Header, body io.Reader) (*http.Response, error) {
+	span, _ := apm.StartSpan(ctx, "Send", "custom")
+	defer span.End()
```

Comment:
Instead of creating these "Send" spans, I think it would be more useful to instrument the underlying libbeat/kibana.Client's HTTP client. This will have a handful of benefits:
 - Slightly more descriptive span names, using the standard HTTP client span naming
 - HTTP span context -- namely status code, but potentially also other things later
 - "external.http" span timing breakdowns (vs. "custom")
 - Trace propagation, enabling distributed tracing from APM Server to the Kibana server

To implement that, in `ConnectingClient.connect`, you can modify the resulting client's `Connection.HTTP`, wrapping it with apmhttp.WrapClient.

EDIT: I just read the PR description properly, and I see you have thought of this and have a PR up for libbeat. I still think it would be worth doing it as described above in the mean time.

Reply:
Agree with all the reasons why instrumenting the HTTP client is better, should have linked to the PR in beats more clearly.

Currently no `context` is passed to libbeat's `client.Send` method, even when the `http.Client` is wrapped, not sure how the spans should be created. I tried it without manual instrumentation in the `Send` method and wrapping the client as suggested in the APM Server code already, but no spans are created. Can you please point me to what I am overlooking here? 

```
func (c *ConnectingClient) Send(ctx context.Context, method, extraPath string, params url.Values,
	headers http.Header, body io.Reader) (*http.Response, error) {
	c.m.RLock()
	defer c.m.RUnlock()
	if c.client == nil {
		return nil, errNotConnected
	}

	return c.client.Send(method, extraPath, params, headers, body)
}

func (c *ConnectingClient) connect() error {
       ...
	client, err := kibana.NewClientWithConfig(c.cfg)
	if err != nil {
		return err
	}
	client.HTTP = apmhttp.WrapClient(client.HTTP)
      ...
```

Reply:
Sorry, I overlooked that you had to update the `Send` method to pass context. What I suggested isn't possible. I'd be inclined to leave these "Send" spans out, and just do the rest once the libbeat change is merged.